### PR TITLE
Additional Windows related updates after Codec2 decoupling

### PR DESCRIPTION
**Pull request by tmiw**
While making sure freedv-gui still compiles, I noticed I was having build issues with MinGW. Further investigation revealed. that `-fvisibility=hidden` doesn't actually work for DLLs, so this PR renames each of the Codec2 functions used to avoid duplicate symbol errors.

See #43 for more details of the original change.

### CODE REVIEW

File: src/from_codec2/kiss_fft.c [hunk 1]
```diff
@@ -12,7 +12,7 @@ Redistribution and use in source and binary forms, with or without modification,
 THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
-
+#include "codec2_renames.h"
```
Comment (tmiw) [inline src/from_codec2/kiss_fft.c]: Unfortunately we'll have to remember to re-add this every time these files get updated. Supposedly we can conditionally make these hidden (see [this](https://gcc.gnu.org/wiki/Visibility)) but that change may be better done in Codec2.

**Comment by drowe67** [comment]
> While making sure freedv-gui still compiles, I noticed I was having build issues with MinGW. Further investigation revealed. that `-fvisibility=hidden` doesn't actually work for DLLs,

Oh dear, that's unfortunate :disappointed: 

Should we remove `-fvisibility=hidden` or revert the `CMakefile.txt` given it's not a global solution?

File: src/from_codec2/kiss_fft.c (hunk 1 above)
Comment (drowe67) [reply src/from_codec2/kiss_fft.c]: I think that's OK.  Hmm, I wonder if it's possible to include a file on the gcc command line, much like we can -D a define :thinking:

Comment (drowe67) [reply src/from_codec2/kiss_fft.c]: How about `-include` https://linux.die.net/man/1/gcc ?

File: src/from_codec2/codec2_renames.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#define hpf __codec2__hpf
+#define dft_speech __codec2__dft_speech
+#define two_stage_pitch_refinement __codec2__two_stage_pitch_refinement
+#define estimate_amplitudes __codec2__estimate_amplitudes
+#define est_voicing_mbe __codec2__est_voicing_mbe
+#define make_synthesis_window __codec2__make_synthesis_window
+#define synthesise __codec2__synthesise
+#define codec2_rand __codec2__codec2_rand
+
+#endif /* CODEC2_RENAMES_H */
```
Comment (drowe67) [inline src/from_codec2/codec2_renames.h]: maybe add a newline here

**Comment by tmiw** [comment]
> > While making sure freedv-gui still compiles, I noticed I was having build issues with MinGW. Further investigation revealed. that `-fvisibility=hidden` doesn't actually work for DLLs,
> 
> Oh dear, that's unfortunate 😞
> 
> Should we remove `-fvisibility=hidden` or revert the `CMakefile.txt` given it's not a global solution?

I think it's still good practice if only so that someone doesn't decide to try to call `__codec2__codec2_fftr()` for example. It's also possible that a different version of Clang or GCC will eventually make that command line option work for Windows too.

(Speaking of which, I've been meaning to build a new Debian based Docker container that uses [the LLVM version of MinGW](https://github.com/mstorsjo/llvm-mingw). On top of a newer compiler, we'd also get the ability to easily build ARM builds for Windows.)

File: src/from_codec2/codec2_renames.h (hunk 2 above)
Comment (tmiw) [reply src/from_codec2/codec2_renames.h]: Done.

File: src/from_codec2/kiss_fft.c (hunk 1 above)
Comment (tmiw) [reply src/from_codec2/kiss_fft.c]: That works. Just pushed a new commit using this.

Comment (drowe67) [reply src/from_codec2/kiss_fft.c]: Cool.  It's quite a neat solution now :+1:

**Comment by drowe67** [comment]
@tmiw OK if I merge this?

**Comment by tmiw** [comment]
> @tmiw OK if I merge this?

Go for it. 👍

**Pull request merged**